src/common/Surface.js

```diff
@@ -130,13 +138,42 @@
             .display(this.showTitle())
             .render()
         ;
+
+        var surfaceButtons = element.selectAll(".surface-button").data(this.buttonAnnotations());
+        surfaceButtons.enter().append("g").classed("surface-button",true)
+            .each(function (button, idx) {
+                context._surfaceButtons[idx] = new Icon()
+                .faChar(button["char"])
```

Comment:
Can you switch to using the (preferable) "." syntax - the main reason being that if we rename one of those attributes, lint/browser will report a syntax error for "." syntax, but this code will silently fail...